File: elixir/504-Base7/main.exs
defmodule Solution do
  def convert_to_base7(0),                 do: "0"
  def convert_to_base7(num) when num < 0,  do: "-" <> convert_to_base7(abs(num))
  def convert_to_base7(num),               do: convert(num, "")

  defp convert(0, acc), do: acc

  defp convert(num, acc) do
    digit = rem(num, 7)
    num = div(num, 7)

    convert(num, "#{digit}" <> acc)
  end
end

IO.inspect(Solution.convert_to_base7(100))
IO.inspect(Solution.convert_to_base7(-7))
IO.inspect(Solution.convert_to_base7(0))

File: cpp/504-Base7/main.cpp
#include <string>
#include <cmath>

using namespace std;

class Solution 
{
public:
    string convertToBase7(int num) 
    {
        auto negative = num < 0;
        auto value = abs(num);
        auto multiplier = 1;
        auto base7 = 0;

        while(value)
        {
            auto digit = value % 7;
            value /= 7;

            base7 += (digit * multiplier);
            multiplier *= 10;
        }

        if(negative) base7 *= -1;

        return to_string(base7);
    }
};

int main()
{
    Solution s;
    auto test1 = s.convertToBase7(100);
    auto test2 = s.convertToBase7(-7);
}